solidity/contracts/token/HypERC20CollateralVaultDeposit.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC4626.sol";
+import {HypERC20Collateral} from "./HypERC20Collateral.sol";
+
+/**
+ * @title Hyperlane ERC20 Token Collateral with deposits collateral to a vault
+ * @author ltyu
+ */
+contract HypERC20CollateralVaultDeposit is HypERC20Collateral {
+    // Address of the ERC4626 compatible vault
+    ERC4626 public immutable vault;
+
+    // Internal balance of total asset deposited
+    uint256 public assetDeposited;
```

Comment:
can we reuse `vault.balanceOf(address(this))` so we dont make this contract stateful?

Reply:
As per discussion. Will keep `assetDeposited`